src/gui/torrentcontentmodel.cpp

```diff
@@ -41,27 +52,113 @@
 
 namespace
 {
+    const QLatin1String useSystemFileIconsSettingKey("Preferences/Advanced/useSystemFileIcons");
+
     QIcon getDirectoryIcon()
     {
         static QIcon cached = GuiIconProvider::instance()->getIcon("inode-directory");
         return cached;
     }
 
-    QIcon getFileIcon()
+    class UnifiedFileIconProvider: public QFileIconProvider
     {
-        static QIcon cached = GuiIconProvider::instance()->getIcon("text-plain");
-        return cached;
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            Q_UNUSED(info);
+            static QIcon cached = GuiIconProvider::instance()->getIcon("text-plain");
+            return cached;
+        }
+    };
+#ifdef Q_OS_WIN
+    // See QTBUG-25319 for explanation why this is required
+    class WinShellFileIconProvider: public UnifiedFileIconProvider
+    {
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            SHFILEINFO sfi = { 0 };
+            HRESULT hr = ::SHGetFileInfo(reinterpret_cast<LPCTSTR>(info.absoluteFilePath().constData()),
```

Comment:
Since we don't support ANSI WinAPI we shouldn't use compatibility layer. Please use SHGetFileInfoW and LPCWSTR directly.
Also your code wrong. You cast 8-bit char array to wchar_t array.